src/analysis/constant_speed.lean

```diff
@@ -88,19 +88,72 @@ lemma has_constant_speed_on_with_iff_variation_on_from_to_eq :
 begin
   split,
   { rintro h, refine ⟨h.has_locally_bounded_variation_on, λ x xs y ys, _⟩,
-    rw has_constant_speed_on_with.iff_ordered at h,
+    rw has_constant_speed_on_with_iff_ordered at h,
     rcases le_total x y with xy|yx,
     { rw [variation_on_from_to.eq_of_le f s xy, h xs ys xy,
           ennreal.to_real_of_real (mul_nonneg l.prop (sub_nonneg.mpr xy))], },
     { rw [variation_on_from_to.eq_of_ge f s yx, h ys xs yx ,
           ennreal.to_real_of_real  (mul_nonneg l.prop (sub_nonneg.mpr yx)),
           mul_comm ↑l, mul_comm ↑l, ←neg_mul, neg_sub], }, },
-  { rw has_constant_speed_on_with.iff_ordered,
+  { rw has_constant_speed_on_with_iff_ordered,
     rintro h x xs y ys xy,
     rw [←h.2 xs ys, variation_on_from_to.eq_of_le f s xy,
         ennreal.of_real_to_real (h.1 x y xs ys)], },
 end
 
+lemma has_constant_speed_on.mono {t : set ℝ} (ts : t ⊆ s)
+  (hts : ∀ ⦃x⦄ (hx : x ∈ t) ⦃y⦄ (hy : y ∈ t), t ∩ (Icc x y) = s ∩ (Icc x y))
+  (hfs : has_constant_speed_on_with f s l) : has_constant_speed_on_with f t l :=
+λ x hx y hy, by rw [←hfs (ts hx) (ts hy), ←hts hx hy]
+
+lemma has_constant_speed_on_union {t : set ℝ} {x : ℝ} (hs : is_greatest s x) (ht : is_least t x)
```

Comment:
dot notation `has_constant_speed_on_with.union`? Since one of your assumptions is of the form `has_constant_speed_on_with`, this makes it more convenient to use. Side remark: if you switch to dot notation, it's better if the assumptions `hs` and `ht` come later because otherwise when using dot notation Lean can get confused. Same thing several times below.